Tools/clinic/clinic.py

```diff
@@ -4666,9 +4666,9 @@ def state_dsl_start(self, line: str) -> None:
                 fail(str(e))
             return
 
-        self.next(self.state_modulename_name, line)
+        self.parse_modulename_name(line)
 
-    def state_modulename_name(self, line: str) -> None:
+    def parse_modulename_name(self, line: str) -> None:
```

Comment:
I renamed this function to distinguish it from the `state_foo` functions, all of which _must_ only ever be called indirectly via `self.next()`

Reply:
What do you think of `parse_function_declaration`?

```suggestion
    def parse_function_declaration(self, line: str) -> None:
```